### ISSUE TITLE
Add screen reader support to the workspace search plugin

### ISSUE THREAD

**Issue by microbit-robert** [google/blockly#10324]
### Check for duplicates

- [x] I have searched for similar issues before opening a new one.

### Component

WorkspaceSearch

### Problem

This plugin has keyboard navigation, but doesn't have screen reader support yet. 

### Request

Add support for screen readers that describes:

- How workspace search works (e.g., it is currently esc to focus the found block and this is not obvious)
- Announce when matching blocks are found (perhaps we have to announce the currently matching block - unclear)
- Is focusing the found block when the user presses Esc enough to understand that the workspace search input has closed and focus has now moved to the block you searched for?
